aconfig/aconfig.py

```diff
@@ -293,6 +324,14 @@ def _env_var_from_key(self, config_key):
         return re.sub(self._search_pattern, '_', config_key.upper())
 
 
+class ImmutableConfig(ImmutableAttributeAccessDict, Config):
+    """This class is the Immutable version of Config"""
+    def __init__(self, config, override_env_vars=True):
+        """See :func:`~aconfig.aconfig.Config.__init__`"""
+        assert isinstance(config, dict)
```

Comment:
I think for robustness, this should be a proper `if not isinstance...` check rather than an `assert`.

Reply:
```suggestion
        if not isinstance(config, dict):
            raise TypeError("config must be a dict")
```

Reply:
ooh, well I copied from the existing init function so we should change that too 😉

Reply:
Ah, I thought that might be the case, but I didn't look carefully enough